### PR TITLE
databases-week1/yuejiao

### PR DESCRIPTION
**Pull request by YuejiaoShi**
Homework Task for databases-week1:

- [x] Find out how many tasks are in the task table
- [x] Find out how many tasks in the task table do not have a valid due date
- [x] Find all the tasks that are marked as done
- [x] Find all the tasks that are not marked as done
- [x] Get all the tasks, sorted with the most recently created first
- [x] Get the single most recently created task
- [x] Get the title and due date of all tasks where the title or description contains database
- [x] Get the title and status (as text) of all tasks
- [x] Get the name of each status, along with a count of how many tasks have that status
- [x] Get the names of all statuses, sorted by the status with most tasks first

[Data used for homework](https://github.com/HackYourFuture-CPH/databases/blob/main/lesson1/lesson-data.sql)

[Homework requirements link](https://github.com/HackYourFuture-CPH/databases/tree/main/lesson1#homework)

### CODE REVIEW

File: databases/week1/homework.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+-- 6. Get the single most recently created task
+SELECT * FROM task ORDER BY created DESC LIMIT 1;
+
+-- 7. Get the title and due date of all tasks where the title or description contains database
+SELECT task.title, task.due_date FROM task WHERE title LIKE '%database%' OR description LIKE '%database%';
+
+-- 8. Get the title and status (as text) of all tasks
+SELECT task.title AS task_title, status.name AS task_status 
+FROM task LEFT JOIN status ON task.status_id = status.id;
```
Comment (NatsGt) [inline databases/week1/homework.sql]: You don't need a left join for this query, a join would be enough (join default is inner join). In this exercise you cant see the difference between them. You can see the main differences here https://www.w3schools.com/mysql/mysql_join.asp

File: databases/week1/homework.sql [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+-- 9. Get the name of each status, along with a count of how many tasks have that status
+SELECT status.name AS task_status, Count(*) AS task_amount
+FROM task RIGHT JOIN status ON task.status_id = status.id
+GROUP BY status.name
+
```
Comment (NatsGt) [inline databases/week1/homework.sql:32]: Same comment as above, you dont need a Right Join.

File: databases/week1/homework.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+--  Answer: 35
+SELECT COUNT(*) AS tasks_amount FROM task;
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+--  Answer: 8
+SELECT COUNT(*) AS tasks_with_valid_due_amount FROM task WHERE due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT task.title AS done_tasks FROM task WHERE status_id = (SELECT id FROM status WHERE name='done'); -- 12 tasks
+
```
Comment (NatsGt) [inline databases/week1/homework.sql:11]: I like your solution with the inner query. However, in this case a Join would be enough.

**Comment by NatsGt** [comment]
Very good job. I like how you added names to most titles. I also liked how you tried using inner queries. I might just suggest you to read a bit more on Joins. Besides that, very good job.

File: databases/week1/homework.sql (hunk 1 above)
Comment (YuejiaoShi) [reply databases/week1/homework.sql]: Thank you so much for your review and suggestions! I've made the changes and will practice more about JOIN on W3schools ❤️

**Pull request merged**